### ISSUE TITLE
Node process hangs before exiting

### ISSUE THREAD

**Issue by srchase** [aws/aws-sdk-js-v3#918]
**Describe the bug**
For some operations, the node process will hang for a few seconds before exiting.

**SDK version number**
@aws-sdk/client-s3@1.0.0-alpha.15

**Is the issue in the browser/Node.js?**
Node

**To Reproduce (observed behavior)**
Run the following script:
```
const { S3 } = require("@aws-sdk/client-s3");

(async () => { 
    const client = new S3({ region: "us-west-2" })
    await client.putObjectTagging({
        Bucket: "bucket",
        Key: "key",
        Tagging: [{ Key: "bar", Value: "foo" }]
    }).then((res) => {
        console.log(res);
    }).catch((err) => {
        console.log(err);
    });
})();
```

**Expected behavior**
Response is returned, node process exits immediately

**Actual behavior**
Response is returned, then node process waits for 3-4 seconds before exiting

**Comment by srchase** [comment]
@AllanFly120 suggests that this may be due to collectStream() not getting called for operations that don't have a document or payload binding.  The node process is waiting for the response stream to close itself since it hasn't been collected already.

**Comment by srchase** [comment]
Holding issue until after clients have been updated.

**Comment by srchase** [comment]
Confirmed fixed with [client-s3@1.0.0-alpha.16](https://www.npmjs.com/package/@aws-sdk/client-s3/v/1.0.0-alpha.16)

Code:
```
const { CloudSearch } = require("@aws-sdk/client-cloudsearch");

const { S3 } = require("@aws-sdk/client-s3");

(async () => { 
    const client = new S3({ region: "us-west-2" })
    await client.putObjectTagging({
        Bucket: "foo",
        Key: "bar",
        Tagging: [{ Key: "baz", Value: "qux" }],
    }).then((res) => {
        console.log(res);
    }).catch((err) => {
        console.log(err);
    });
})();
```

The output returns and the process exits immediately.
